packages/gatsby-transformer-asciidoc/src/gatsby-node.js

```diff
@@ -0,0 +1,39 @@
+// const Promise = require(`bluebird`)
+// const _ = require(`lodash`)
+const asciidoc = require(`asciidoctor.js`)()
+// import asciidoc from 'asciidoctor.js'
+
+async function onCreateNode(
+  { node, actions, loadNodeContent, createNodeId, createContentDigest },
+  options
+) {
+  // Filter out non-adoc content
+  if (!node.extension || node.extension !== `adoc`) {
+    return
+  }
+
+  const { createNode, createParentChildLink } = actions
+  // Load Asciidoc contents
+  const content = await loadNodeContent(node)
+  const html = asciidoc.convert(content)
+  
+  const asciiNode = {
+    id: createNodeId(`${node.id} >>> ASCIIDOC`),
+    parent: node.id,
+    internal: {
+      type: `Asciidoc`,
+      mediaType: `text/html`,
+      content: html,
+    },
+    children: [],
+    html,
+    relativePath: node.relativePath,
```

Comment:
Is this needed? I would like to avoid file properties spilling into children node - you can query for parent and get relativePath from there